hearts/tests/test_hearts.py

```diff
@@ -163,18 +163,19 @@ def test_is_valid_follow():
 
 
 def test_lead_the_trick():
-    test_round = Round(PLAYER_LIST)
-    fake_hand = Hand.deserialize(['7d', '6h', 'Ah', '2s'])
-    fake_trick = Trick([(P1, Card('2', 'c'))])
-    test_round.hands[P4] = fake_hand
-    test_round.turn_counter = 3
-    test_round.tricks.append(fake_trick)
-
-    test_round.lead_the_trick(P4, Card('2', 's'))
-    assert test_round.tricks[-1] == Trick([(P4, Card('2', 's'))])
+    round1 = Round(PLAYER_LIST)
+    hand = Hand.deserialize(['7d', '6h', 'Ah', '2s'])
+    trick = Trick([(P1, Card('2', 'c'))])
+    round1.hands[P4] = hand
+    round1.turn_counter = 3
+    round1.tricks.append(trick)
+
+    round1.lead_the_trick(P4, Card('2', 's'))
```

Comment:
Yes, `lead_the_trick` will not be called unless the previous trick is complete.  I think it's safe to assume we won't use it anywhere else.  
`    def play_card(self, player, card):
        if self.is_player_turn(player):
            if len(self.tricks) == 0 or len(self.tricks[-1]) == 4:
                self.lead_the_trick(player, card)`